pkg/scheduler/service/cleaner.go

```diff
@@ -61,3 +125,99 @@ func (c *cleaner) purgeReconciliations(transition *ClusterStatusTransition, conf
 		}
 	}
 }
+
+func (c *cleaner) getLatestReconciliations(latestEntitiesCount int, transition *ClusterStatusTransition) ([]*model.ReconciliationEntity, error) {
+	return transition.ReconciliationRepository().GetReconciliations(&reconciliation.Limit{Count: latestEntitiesCount})
+}
+
+func (c *cleaner) dropRecordsOlderThan(t time.Time, transition *ClusterStatusTransition) error {
+	deadline := t.UTC()
+	list, err := c.findReconciliationsOlderThan(deadline, transition)
+	if err != nil {
+		return err
+	}
+
+	c.logger.Infof("[CLEANER] Removing records older than: %s. Records found: %d", deadline.String(), len(list))
+	if len(list) > 0 {
+		c.removeReconciliations(list, transition, func(m *model.ReconciliationEntity) bool { return true })
+	}
+	return err
+}
+
+func (c *cleaner) dropSuccessfulRecordsOlderThan(t time.Time, transition *ClusterStatusTransition) error {
+	deadline := t.UTC()
+	list, err := c.findReconciliationsOlderThan(deadline, transition)
+	if err != nil {
+		return err
+	}
+
+	c.logger.Infof("[CLEANER] Removing successful records older than: %s. Records found: %d (only some must be removed)", deadline.String(), len(list))
+	if len(list) > 0 {
+		c.removeReconciliations(list, transition, func(m *model.ReconciliationEntity) bool { return m.Status.IsFinalStable() })
+	}
+	return err
+}
+
+func (c *cleaner) findReconciliationsOlderThan(t time.Time, transition *ClusterStatusTransition) ([]*model.ReconciliationEntity, error) {
+	reconciliations, err := transition.ReconciliationRepository().GetReconciliations(&reconciliation.WithCreationDateBefore{
+		Time: t,
+	})
+	if err != nil {
+		c.logger.Errorf("Cleaner failed to get reconciliations older than %s: %s", t.String(), err.Error())
+		return nil, err
+	}
+	return reconciliations, nil
+}
+
+func (c *cleaner) removeReconciliations(list []*model.ReconciliationEntity, transition *ClusterStatusTransition, filter func(*model.ReconciliationEntity) bool) {
+	cnt := 0
+	for i := range list {
+		id := list[i].SchedulingID
+		//TODO: does this mean "successful" ?
```

Comment:
is this statement clear now? we can remove the TODO.

Reply:
Nope, it's not :( I don't know, have to ask Tobias.

Reply:
OK, I have the confirmation the condition is correct.

Reply:
Removed